x-pack/test/observability_ai_assistant_api_integration/tests/knowledge_base/knowledge_base.spec.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { SUPPORTED_TRAINED_MODELS } from '../../../functional/services/ml/api';
+
+interface KnowledgeBaseEntry {
+  id: string;
+  text: string;
+}
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const ml = getService('ml');
+  const es = getService('es');
+  const TINY_ELSER = {
+    ...SUPPORTED_TRAINED_MODELS.TINY_ELSER,
+    id: SUPPORTED_TRAINED_MODELS.TINY_ELSER.name,
+  };
+  const KNOWLEDGE_BASE_API_URL = `/internal/observability_ai_assistant/kb`;
+  const KB_INDEX = '.kibana-observability-ai-assistant-kb-*';
+
+  describe('Knowledge base', () => {
+    before(async () => {
+      const config = {
+        ...ml.api.getTrainedModelConfig(TINY_ELSER.name),
+        input: {
+          field_names: ['text_field'],
+        },
+      };
+      await ml.api.importTrainedModel(TINY_ELSER.name, TINY_ELSER.id, config);
```

Comment:
what happens when we call setup before importing the model via the ML API?

Reply:
It doesn't work because I need to send in this config:
https://github.com/elastic/kibana/pull/189000/files#diff-3d95f9db91b6d33b6df4ae983882f0821a9d3ae155b3be9d82ff7d08b2fe8cc9R18

Which loads this file:
https://github.com/elastic/kibana/blob/main/x-pack/test/functional/services/ml/resources/trained_model_definitions/pt_tiny_elser/config.json

I am assuming something like this is happening already for elser2 as its already listed in the list of ML trained models when kibana starts?

Reply:
ok, yeah I guess ELSER is a built-in model and Tiny ELSER isn't - which makes sense. Maybe we can do this in the setup/teardown functions for the entire test suite instead of per test or test group, to make it behave more similarly to ELSER? I don't really have a strong preference for either.